client/lib/query-args/test/index.js

```diff
@@ -0,0 +1,74 @@
+import page from 'page';
+import useQueryArgs from '../index';
+
+jest.mock( 'react' );
+jest.mock( 'page' );
+
+describe( '#useQueryArgs', () => {
+	const queryArgs = useQueryArgs();
+
+	global.window = {
+		location: {
+			href: '',
+		},
+	};
+
+	test( 'should remove the query if search is empty', () => {
+		global.window.location.pathname = '/read/search';
+		global.window.location.href = 'https://wordpress.com/read/search?q=test';
+		const expectedResult = '/read/search';
+		queryArgs.setQueryArgs();
+
+		expect( page.replace ).toBeCalledWith( expectedResult );
+	} );
+
+	test( 'should navigate to original url if there is no search', () => {
+		global.window.location.pathname = '/plugins';
+		global.window.location.href = 'https://wordpress.com/plugins';
+		queryArgs.setQueryArgs();
+
+		expect( page.replace ).toBeCalledWith( '/plugins' );
+	} );
+
+	test( 'should replace current query with new one even when using custom query key', () => {
+		global.window.location.pathname = '/read/search';
+		global.window.location.href = 'https://wordpress.com/read/search?q=test';
+		const expectedResult = '/read/search?q=reader+is+super+awesome';
+		const newSearchTerm = 'reader is super awesome';
+
+		queryArgs.setQueryArgs( { q: newSearchTerm } );
+
+		expect( page.replace ).toHaveBeenLastCalledWith( expectedResult );
+	} );
+
+	test( 'should navigate to a page without a query search', () => {
+		global.window.location.pathname = '/';
+		global.window.location.href = 'https://wordpress.com';
+		queryArgs.setQueryArgs();
+
+		expect( page.replace ).toBeCalledWith( '/' );
+	} );
+
+	test( 'should replace page url on a query search', () => {
+		global.window.location.pathname = '/';
+		const baseUrl = 'https://wordpress.com';
+		global.window.location.href = baseUrl + '?s=test';
+
+		const newSearchTerm = 'replaced-search-term';
+		const expectedUrl = '/?s=' + newSearchTerm;
+
+		queryArgs.setQueryArgs( { s: newSearchTerm } );
+
+		expect( page.replace ).toHaveBeenLastCalledWith( expectedUrl );
+	} );
+
+	test( 'searchTerm should contain the last searchTerm performed on setQueryArgs', () => {
+		const baseUrl = 'https://wordpress.com/plugins';
+		const newSearchTerm = 'replaced-search-term';
+		global.window.location.pathname = '/';
+		global.window.location.href = baseUrl + '?s=' + newSearchTerm;
+
+		queryArgs.setQueryArgs( newSearchTerm );
+		expect( queryArgs.getQueryArgs() ).toEqual( { s: 'replaced-search-term' } );
+	} );
```

Comment:
This test is a bit broken, the assertion is only checking that `s` was untouched while quite a bit more is happening. Though I think we should just not support this feature and can remove the test.

```suggestion
	test( 'searchTerm should contain the last searchTerm performed on setQueryArgs', () => {
		global.window.location.href = 'https://wordpress.com/plugins?s=search-term';
		queryArgs.setQueryArgs( 'replaced-search-term' );
		expect( page.replace ).toHaveBeenLastCalledWith( '/plugins?s=replaced-search-term' );
	} );
```
```
Expected: "/plugins?s=replaced-search-term"
    Received
           6: "/?s=replaced-search-term"
    ->     7: "/plugins?s=search-term&0=r&1=e&2=p&3=l&4=a&5=c&6=e&7=d&8=-&9=s&10=e&11=a&12=r&13=c&14=h&15=-&16=t&17=e&18=r&19=m"
```